src/odemis/gui/cont/acquisition.py

```diff
@@ -927,6 +928,203 @@ def on_file_export_done(self, future):
+            # sharing the label with the estimated time-to-completion).
+            if self._main_data_model.is_acquiring.value:
+                return
+            # Display acquisition time
+            projects = self._tab_data_model.projects.value
+            acq_time = fastem.estimateTime([roa for p in projects for roa in p.roas.value])
+            acq_time = math.ceil(acq_time)  # round a bit pessimistic
+            txt = u"Estimated time is {}."
+            txt = txt.format(units.readable_time(acq_time))
+        logging.debug("Updating status message %s, with level %s", txt, lvl)
+        self.lbl_acqestimate.SetLabel(txt)
+        self._show_status_icons(lvl)
+
+    def _get_undefined_calibrations(self):
+        """
+        returns (list of str): names of ROCs which are undefined
+        """
+        undefined = set()
+        for p in self._tab_data_model.projects.value:
+            for roa in p.roas.value:
+                roc = roa.roc.value
+                if roc.coordinates.value == stream.UNDEFINED_ROI:
+                    undefined.add(roc.name.value)
+        return sorted(undefined)
+
+    def update_roa_count(self):
+        roas = [roa for p in self._tab_data_model.projects.value for roa in p.roas.value]
+        self.txt_num_rois.SetValue("%s" % len(roas))
+        self.roa_count = len(roas)
+
+    def _show_status_icons(self, lvl):
+        # update status icon to show the logging level
+        self.bmp_acq_status_info.Show(lvl in (logging.INFO, logging.DEBUG))
+        self.bmp_acq_status_warn.Show(lvl == logging.WARN)
+        self._tab_panel.Layout()
+
+    def _reset_acquisition_gui(self, text=None, level=None):
+        """
+        Set back every GUI elements to be ready for the next acquisition
+        text (None or str): a (error) message to display instead of the
+          estimated acquisition time
+        level (None or logging.*): logging level of the text, shown as an icon.
+          If None, no icon is shown.
+        """
+        self.btn_cancel.Hide()
+        self.btn_acquire.Enable()
+        self._tab_panel.Layout()
+
+        if text is not None:
+            self.lbl_acqestimate.SetLabel(text)
+            self._show_status_icons(level)
+        else:
+            self.update_acquisition_time()
+
+    def on_acquisition(self, evt):
+        """
+        Start the acquisition (really)
+        """
+        self._main_data_model.is_acquiring.value = True
+        self.btn_acquire.Enable(False)
+        self.btn_cancel.Enable(True)
+        self.btn_cancel.Show()
+        self.gauge_acq.Show()
+        self._show_status_icons(None)
+
+        self.gauge_acq.Range = self.roa_count
+        self.gauge_acq.Value = 0
+
+        # Acquire ROAs for all projects
+        for p in self._tab_data_model.projects.value:
+            ppath = os.path.join(self.path, p.name.value)
+            for roa in p.roas.value:
+                fn = os.path.join(ppath, roa.name.value)
```

Comment:
Is there a place where you ensure that projects have different names and that RoAs of a project have different names? Otherwise the user could end up with loosing data if mistakenly picking up the same name.

Reply:
That is a very good question. For now, Sabrina and I agreed to worry about this in the second iteration. There's two potential ways we can get a conflict:
* using the same name during the same session --> that's easy to check, we just need to go through all the projects/rois in the model
* using the same name after restarting odemis on the same day (will use same external folder) --> more complicated, we need to check external storage, only possible if that is mounted

Then, there is also the question when to check. We could do it just before the acquisition and just add a counter if necessary. Another option would be to check immediately after typing in the GUI and give the user some feedback that the name has changed (if we do it automatically)/ that it has to be changed (if we require the user to do it).

Reply:
I'm implementing a simple check now every time the user changes the ROI/project name and adding a counter if necessary.